_source/_posts/2022-07-12-hosted-auth-cli.md

```diff
@@ -0,0 +1,159 @@
+---
+layout: blog_post
+title: "Get Started with the Okta CLI and Hosted Authentication in Python"
+author: erikka-innes
+by: contractor
+communities: [python]
+description: "Learn how to set up the Okta CLI and run the hosted authentication sample application."
+tags: []
+tweets:
+- "Configure the Okta CLI and run your first sample application using hosted authentication!"
+- "Try the Okta CLI, the easiest way to create your first Okta application with hosted authentication."
+- ""
+image:
+type: conversion
+---
+
+Okta is a customizable, secure solution that lets you add authentication and access management to your applications. Since this can be a complex topic, we recommend setting up your first project with our CLI. In this walk through, we will:
+- Configure the Okta CLI
+- Install the Okta Python SDK
+- Teach you how to create your first application
+- Learn about authentication and authorization credentials
+- Set up Hosted Authentication with Okta and try it out 
+- Cover troubleshooting issues you might encounter
+
+Ready? Let's get started! 
+
+**Prerequisites**
+
+- Okta developer account - these instructions assume you already have signed up. If you didn't, you can sign up for an account using the CLI! 
+- Python
+
+We will go over how to install the CLI and SDK during this tutorial. 
+
+{% include toc.md %}
+
+## Overview concepts
+This section quickly goes over a few key concepts.
+
+### Org URL
+When you're starting out with Okta, you're provided an Org URL. This is what you'll use to set up all of your applications. Think of it as a reference ID that lets Okta figure out who you are and which projects are yours. If you want to learn more about the complexities of the Org URL (it has many more features than what we've just described and it's possible to have more than one depending on your needs), you can read about it here: https://developer.okta.com/docs/concepts/okta-organizations/#org-features
+
+### Tokens, client IDs, and client secrets
+You're also provided some information for authentication. It can get a bit confusing as there are tokens, client IDs, and client secrets. You need to use all three for the Okta application we're working on today. Let's quickly go over what each thing does for you. 
+Tokens - these are like an API key. You're going to provide the token so that you can authenticate with the API. You use this for every request and it contains details about what you can and can't access. They are presented to you only once, and never shown again. However, if you forget your token but you authenticate with the CLI, you can retrieve it. We'll go over that in a bit. 
```

Comment:
```suggestion
- Tokens - these are like an API key. You're going to provide the token so that you can authenticate with the API. You use this for every request and it contains details about what you can and can't access. Tokens are presented to you only once, and never shown again. However, if you forget your token but you authenticate with the CLI, you can retrieve it. We'll go over that in a bit. 
```
bulleted list for easier scanning